SofaKernel/modules/SofaDeformable/PolynomialSpringsForceField.inl

```diff
@@ -0,0 +1,598 @@
+            msg_info() << "Spring length value: " << m_springLength[i];
+            if (m_computeSpringsZeroLength[i] == 1) {
+                m_initialSpringLength[i] = m_springLength[i];
+                msg_info() << "Spring zero length: " << m_initialSpringLength[i];
+                m_computeSpringsZeroLength[i] = 0;
+            }
+            m_weightedCoordinateDifference[i] = m_weightedCoordinateDifference[i] / m_springLength[i];
+
+            m_strainValue[i] = std::fabs(m_springLength[i] - m_initialSpringLength[i]) / m_initialSpringLength[i];
+            double forceValue = PolynomialValue(i, m_strainValue[i]);
+            m_strainSign[i] = m_springLength[i] - m_initialSpringLength[i] >= 0 ? 1.0 : compressionValue;
+            msg_info() << "Strain sign: " << m_strainSign[i];
+            msg_info() << "Strain value: " << m_strainValue[i];
+            msg_info() << "Force value: " << forceValue;
+
+            f1[firstIndex] += forceValue * m_strainSign[i] * m_weightedCoordinateDifference[i];
+            f2[secondIndex] -= forceValue * m_strainSign[i] * m_weightedCoordinateDifference[i];
+            msg_info() << "Applied force value: " << forceValue * m_strainSign[i] * m_weightedCoordinateDifference[i];
+
+            ComputeJacobian(i, i);
+        }
+    }
+}
+
+
+
+template<class DataTypes>
+void PolynomialSpringsForceField<DataTypes>::ComputeJacobian(unsigned int stiffnessIndex, unsigned int springIndex)
+{
+    msg_info() << "\n\nCompute derivative: ";
+    msg_info() << "spring length: " << m_springLength[springIndex];
+    // Compute stiffness dF/dX for nonlinear case
+
+    msg_info() << "weighted difference: " << m_weightedCoordinateDifference[springIndex][0] << " "
+               << m_weightedCoordinateDifference[springIndex][1] << " " << m_weightedCoordinateDifference[springIndex][2];
+
+    // compute polynomial result
+    double polynomialForceRes = PolynomialValue(stiffnessIndex, m_strainValue[springIndex]) / m_springLength[springIndex];
+    msg_info() << "PolynomialForceRes: " << polynomialForceRes;
+
+    double polynomialDerivativeRes = PolynomialDerivativeValue(stiffnessIndex, m_strainValue[springIndex]) / m_initialSpringLength[springIndex];
+    msg_info() << "PolynomialDerivativeRes: " << polynomialDerivativeRes;
+
+    // compute data for Jacobian matrix
+    JacobianMatrix& jacobMatrix = m_differential[springIndex];
+    for(unsigned int firstIndex = 0; firstIndex < m_dimension; firstIndex++)
+    {
+        for(unsigned int secondIndex = 0; secondIndex < m_dimension; secondIndex++)
+        {
+            jacobMatrix[firstIndex][secondIndex] = (polynomialDerivativeRes - polynomialForceRes) *
+                    m_weightedCoordinateDifference[springIndex][firstIndex] * m_weightedCoordinateDifference[springIndex][secondIndex];
+        }
+        jacobMatrix[firstIndex][firstIndex] += polynomialForceRes;
+    }
+
+    for(unsigned int firstIndex = 0; firstIndex < m_dimension; firstIndex++)
```

Comment:
The msg_info is done in the for loop... so if there is no printing we are still iterating.
Add a condition like this one:  if(notMuted()){ the loops... }